### PR TITLE
Store route blob ID on the DHT and manage route IDs

### PR DESCRIPTION
**Pull request by tripledoublev**
- Added DHT schema support for storing route blob IDs with multiple subkeys.
- Integrated group route ID management to handle messaging through VeilidAPI.
- Implemented functionality for creating and storing custom private routes in the DHT.
- Enhanced test coverage to verify route creation, message sending, and receiving via AppMessage updates.

### CODE REVIEW

File: src/backend.rs [hunk 1]
```diff
@@ -111,7 +106,13 @@ impl Backend {
 
         println!("Waiting for network ready state");
 
-        rx.recv().await.expect("Unable to wait for veilid init");
+        let update_rx = Arc::new(Mutex::new(update_rx));
```
Comment (RangerMauve) [inline src/backend.rs]: If we want multiple consumers we should use a broadcast channel instead.

File: src/common.rs [hunk 2]
```diff
@@ -90,6 +100,88 @@ pub trait DHTEntity {
...
     }
 
+     // Common methods for DHT operations
+    async fn store_route_id_in_dht(&self, subkey: u32, route_id_blob: Vec<u8>) -> Result<()> {
+        let routing_context = &self.get_routing_context();
+    
+        println!("Attempting to open DHT record...");
+    
+        // Open the DHT record using the group's DHT record
+        let dht_record = routing_context
```
Comment (RangerMauve) [inline src/common.rs]: Is this a noop if it's called multiple times? Feels like we should only open the dht record when initializing the repo. That needs to have happened for `self.get_dht_record` to work right?

File: src/common.rs [hunk 3]
```diff
@@ -90,6 +100,88 @@ pub trait DHTEntity {
...
+            .await
+            .map_err(|e| {
+                println!("Failed to set value in DHT: {}", e);
+                anyhow!("Failed to set value in DHT: {}", e)
+            })?;
+    
+        println!("Value set successfully in DHT.");
+    
+        // Close the DHT record after setting the value
+        routing_context.close_dht_record(self.get_dht_record().key().clone()).await?;
```
Comment (RangerMauve) [inline src/common.rs]: Maybe we should only close when the repo gets closed?

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: src/backend.rs (hunk 1 above)
Comment (tripledoublev) [reply src/backend.rs]: Switched to broadcast channel

File: src/common.rs (hunk 2 above)
Comment (tripledoublev) [reply src/common.rs]: Removed open and close dht for  both store route id and get route id functions

File: src/common.rs (hunk 3 above)
Comment (tripledoublev) [reply src/common.rs]: Your comment made me realize we do not have a close_repo function yet.

File: src/common.rs [hunk 4]
```diff
@@ -90,6 +100,63 @@ pub trait DHTEntity {
...
+        Ok(stored_blob.data().to_vec())
+    }
+    
+
+    // Send an AppMessage to the repo owner using the stored route ID blob
+    async fn send_message_to_owner(
+        &self,
+        veilid: &VeilidAPI,
+        message: Vec<u8>,
+        subkey: u32,
```
Comment (RangerMauve) [inline src/common.rs:143]: Lets hardcode this key in a const and set it to `2`

File: src/lib.rs [hunk 5]
```diff
@@ -76,7 +75,7 @@ mod tests {
         assert_eq!(name, repo_name);
 
         // Add repo to group
-        loaded_group.add_repo(repo).await.expect("Unable to add repo to group");
+        loaded_group.add_repo(repo.clone()).await.expect("Unable to add repo to group");
```
Comment (RangerMauve) [inline src/lib.rs:78]: I think instead of adding a repo groups should be able to load a repo from a DHT key + encryption key or create a new repo

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 5 above)
Comment (tripledoublev) [reply src/lib.rs:78]: I believe what you are describing is covered by the `get_repo` method

File: src/common.rs (hunk 4 above)
Comment (tripledoublev) [reply src/common.rs:143]: 071b40637fffbda26ef116c304acfb201186f964

**Review by RangerMauve — APPROVED** [review]

**Pull request merged**